### PR TITLE
Add feature `getrandom`

### LINKED ISSUE #85: Reexport rand_core
There are issues with conflicting versions of `rand_core` since x25519 does not use the latest version. It would be helpful if `rand_core` was exported to avoid conflicts

### PR DESCRIPTION
**Pull request by pinkforest**
Fixes #85 
Supercedes #103 by re-rolling the commit with Co-authored-by @cipriancraciun (it was targeting main branch)

Provides the new feature `getrandom` with convenience `random()` functions (naming aligned to other daleks)

### CODE REVIEW

**Review by tarcieri — APPROVED** [review]

**Review by tarcieri — CHANGES_REQUESTED** [review]
Weird, what's up with the CI failures?

**Comment by pinkforest** [comment]
Because the fix CI PR has not been merged

**Comment by pinkforest** [comment]
Now it's failing pending this:
- https://github.com/dalek-cryptography/x25519-dalek/pull/115

We can merge that first pls and then rebase and merge this :)

**Comment by pinkforest** [comment]
Oh the minor doc nit fix was actually in #112 but anyways I fixed it here as well - should merge ok now.

**Review by tarcieri — APPROVED** [review]
I'll re-approve but I do wonder about the names `EphemeralSecret::new` vs `::random`.

Perhaps rename `::new` to `::new_from_rng` and call the `OsRng`-based one `new`? Or `::random_from_rng` and `::random`?

**Comment by pinkforest** [comment]
I worry about people getting confused if we re-use `::new` so the 2nd makes sense - I'll update it

**Comment by pinkforest** [comment]
Done it's `::random_from_rng(OsRng)` and `::random()` now :)

**Comment by tarcieri** [comment]
Works for me

File: src/x25519.rs [hunk 1]
```diff
@@ -76,14 +76,19 @@ impl EphemeralSecret {
         SharedSecret(self.0 * their_public.0)
     }
 
-    /// Generate an x25519 [`EphemeralSecret`] key.
-    pub fn new<T: RngCore + CryptoRng>(mut csprng: T) -> Self {
```
Comment (tarcieri) [inline src/x25519.rs:80]: It might be good to retain `new` with a `#[deprecated]` attribute to help people upgrade

Comment (pinkforest) [reply src/x25519.rs:80]: Good idea. btw - should we take random* out from `StaticSecret` as I've seen people using it just to ECDH when they should be using `EphemeralSecret` really.

Comment (pinkforest) [reply src/x25519.rs:80]: Done - I've added `#[deprecated]` agains 2.0.0 with indication of removal in 2.1.0

File: src/x25519.rs [hunk 2]
```diff
@@ -76,14 +76,33 @@ impl EphemeralSecret {
...
+    )]
     pub fn new<T: RngCore + CryptoRng>(mut csprng: T) -> Self {
         let mut bytes = [0u8; 32];
 
         csprng.fill_bytes(&mut bytes);
 
         EphemeralSecret(Scalar::from_bits_clamped(bytes))
     }
+
+    /// Generate an x25519 [`EphemeralSecret`] key with the supplied [`rand_core::OsRng`]
```
Comment (rozbb) [inline src/x25519.rs:92]: Ditto. This is repeated 4 or 5 more times, so I'll just omit the dittos

File: src/x25519.rs [hunk 3]
```diff
@@ -76,14 +76,33 @@ impl EphemeralSecret {
         SharedSecret(self.0 * their_public.0)
     }
 
-    /// Generate an x25519 [`EphemeralSecret`] key.
+    /// Generate an x25519 [`EphemeralSecret`] key with the supplied [`rand_core::OsRng`]
```
Comment (rozbb) [inline src/x25519.rs:79]: Not `OsRng`, but `RngCore`

**Review by rozbb — CHANGES_REQUESTED** [review]
Thank you!

File: src/x25519.rs (hunk 2 above)
Comment (pinkforest) [reply src/x25519.rs]: Yeah I just took it out and simplified it since the rendered documentation links them anyways.

**Comment by pinkforest** [comment]
@tarcieri These PR's got closed because release/2.0 branch got deleted and can't be re-opened.

**Pull request closed without merging**